src/clients/repositoryinfoclient.ts

```diff
@@ -0,0 +1,132 @@
+import { Logger } from "../helpers/logger";
+import { RepoUtils } from "../helpers/repoutils";
+import { IRepositoryContext, RepositoryType } from "../contexts/repositorycontext";
+import { TeamServicesApi } from "./teamservicesclient";
+import { RepositoryInfo } from "../info/repositoryinfo";
+import { TeamProject, TeamProjectCollection } from "vso-node-api/interfaces/CoreInterfaces";
+
+export class RepositoryInfoClient {
+    private _handler: VsoBaseInterfaces.IRequestHandler;
+    private _repoContext: IRepositoryContext;
+
+    constructor(context: IRepositoryContext, handler: VsoBaseInterfaces.IRequestHandler) {
+        this._repoContext = context;
+        this._handler = handler;
+    }
+
+    public async GetRepositoryInfo(): Promise<RepositoryInfo> {
+        let repoInfo: any;
+        let repositoryInfo: RepositoryInfo;
+        let repositoryClient: TeamServicesApi;
+
+        if (this._repoContext.Type === RepositoryType.GIT) {
+            Logger.LogDebug(`Getting repository information for a Git repository at ${this._repoContext.RemoteUrl}`);
+            repositoryClient = new TeamServicesApi(this._repoContext.RemoteUrl, [this._handler]);
+            //TODO: This can throw
+            repoInfo = await repositoryClient.getVstsInfo();
+            repositoryInfo = new RepositoryInfo(repoInfo);
+            return repositoryInfo;
+        } else if (this._repoContext.Type === RepositoryType.TFVC) {
+            Logger.LogDebug(`Getting repository information for a TFVC repository at ${this._repoContext.RemoteUrl}`);
+            //For TFVC, the teamProjectName is retrieved by tf.cmd and set on the context
+            let teamProjectName: string = this._repoContext.TeamProjectName;
+            repositoryInfo = new RepositoryInfo(this._repoContext.RemoteUrl);
+
+            let serverUrl: string;
+            let collectionName: string;
+            if (RepoUtils.IsTeamFoundationServicesRepo(this._repoContext.RemoteUrl)) {
+                // The Team Services collection is ALWAYS defaultCollection, and both the url with defaultcollection
+                // and the url without defaultCollection will validate just fine. However, it expects you to refer to
+                // the collection by the account name. So, we just need to grab the account name and use that to
+                // recreate the url.
+                // If validation fails, we return false.
+                collectionName = repositoryInfo.Account;
+                serverUrl = `https://${repositoryInfo.Account}.visualstudio.com/`;
+                //repositoryClient = new TeamServicesApi(serverUrl, [this._handler]);
+                let valid: boolean = await this.validateTfvcCollectionUrl(serverUrl, this._handler);
+                if (!valid) {
+                    return undefined; //not what we expected (which is a tfvc team services repo)
+                }
+            } else {
+                //TODO: on-prem TFS...?!
```

Comment:
Correct.  This is the first PR for the basic integration (and I used a Team Services repository for that).  On-Prem will be forthcoming.